modules/google-analytics/classes/wp-google-analytics-universal.php

```diff
@@ -195,23 +199,193 @@ public function loop_add_to_cart() {
+		}
+
+		global $product, $woocommerce_loop;
+		$product_sku_or_id = Jetpack_Google_Analytics_Utils::get_product_sku_or_id( $product );
+
+		$item_details = array(
+			'id' => $product_sku_or_id,
+			'name' => $product->get_title(),
+			'category' => Jetpack_Google_Analytics_Utils::get_product_categories_concatenated( $product ),
+			'list' => $list,
+			'position' => $woocommerce_loop['loop']
+		);
+		wc_enqueue_js( "ga( 'ec:addImpression', " . wp_json_encode( $item_details ) . " );" );
+	}
+
+	public function listing_click() {
+		if ( ! Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+			return;
+		}
+
+		if ( ! Jetpack_Google_Analytics_Options::track_product_clicks_is_enabled() ) {
+			return;
+		}
+
+		if ( isset( $_GET['s'] ) ) {
+			$list = "Search Results";
+		} else {
+			$list = "Product List";
+		}
+
+		global $product, $woocommerce_loop;
+		$product_sku_or_id = Jetpack_Google_Analytics_Utils::get_product_sku_or_id( $product );
+
+		$selector = ".products .post-" . esc_js( $product->get_id() ) . " a";
+
+		$item_details = array(
+			'id' => $product_sku_or_id,
+			'name' => $product->get_title(),
+			'category' => Jetpack_Google_Analytics_Utils::get_product_categories_concatenated( $product ),
+			'position' => $woocommerce_loop['loop']
+		);
+
+		wc_enqueue_js(
+			"$( '" . esc_js( $selector ) . "' ).click( function() {
+				if ( true === $( this ).hasClass( 'add_to_cart_button' ) ) {
+					return;
+				}
+
+				ga( 'ec:addProduct', " . wp_json_encode( $item_details ) . " );
+				ga( 'ec:setAction', 'click', { list: '" . esc_js( $list ) . "' } );
+				ga( 'send', 'event', 'UX', 'click', { list: '" . esc_js( $list ) . "' } );
 			} );"
 		);
 	}
 
-	public function wp_footer() {
+	public function product_detail() {
+		if ( ! Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+			return;
+		}
+
+		if ( ! Jetpack_Google_Analytics_Options::track_product_detail_view_is_enabled() ) {
+			return;
+		}
+
+		global $product;
+		$product_sku_or_id = Jetpack_Google_Analytics_Utils::get_product_sku_or_id( $product );
+
+		$item_details = array(
+			'id' => $product_sku_or_id,
+			'name' => $product->get_title(),
+			'category' => Jetpack_Google_Analytics_Utils::get_product_categories_concatenated( $product ),
+			'price' => $product->get_price()
+		);
+		wc_enqueue_js(
+			"ga( 'ec:addProduct', " . wp_json_encode( $item_details ) . " );" .
+			"ga( 'ec:setAction', 'detail' );"
+		);
+	}
+
+	public function checkout_process() {
+		if ( ! Jetpack_Google_Analytics_Options::enhanced_ecommerce_tracking_is_enabled() ) {
+			return;
+		}
+
+		if ( ! Jetpack_Google_Analytics_Options::track_checkout_started_is_enabled() ) {
+			return;
+		}
+
+		$universal_commands = array();
+		$cart = WC()->cart->get_cart();
+
+		foreach ( $cart as $cart_item_key => $cart_item ) {
+			$product = apply_filters( 'woocommerce_cart_item_product', $cart_item['data'], $cart_item, $cart_item_key );
```

Comment:
Could you add a docblock here so as to not trip the parser? Since it's a Woocommerce filter something like `/** This filter is already documented in xxxx.php */` would do the trick.

Reply:
Fixed in https://github.com/Automattic/jetpack/pull/8218/commits/5977a8bd5b8a6aa9ba708b0ed840e838e2a59fcd